tests/ert/unit_tests/config/test_parameter_config.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import re
+from pathlib import Path
+
+import xtgeo
+from surfio import IrapSurface
+
+from ert.config import ExtParamConfig, Field, GenKwConfig, SurfaceConfig
+from ert.field_utils import Shape
+
+
+def assert_parameters_in_logs(
+    expected_provided_parameters: set, expected_defaulted_parameters: set, caplog
+):
+    def logged_properties_to_set(logged_properties: str):
+        # The order of logged properties is arbitrary,
+        # creating a set of the properties solves this
+        return set(logged_properties.strip("'").split("', '"))
+
+    match = re.search(PARAMETER_LOG_PATTERN, caplog.text)
+    provided_parameters = match.group(1)
+    unprovided_parameters = match.group(2)
+    assert logged_properties_to_set(provided_parameters) == expected_provided_parameters
+    assert (
+        logged_properties_to_set(unprovided_parameters) == expected_defaulted_parameters
+    )
+
+
+PARAMETER_LOG_PATTERN = (
+    r"Parameters with input values:\n\{([^}]+)\}\n"
+    r"Parameters with defaulted values:\n\{([^}]+)\}"
+)
+
+
+def test_parameters_are_logged_for_gen_kw_instances(caplog):
```

Comment:
Could we add parameterize here for checking `update=True/False` is logged correctly or just maybe add two parameters in the test?